ui/src/store/account.js

```diff
@@ -53,6 +54,9 @@ export default {
 		},
 		setEmail(state, { email }) {
 			state.email = email;
+		},
+		setLoggingIn(state, value) {
+			state.loggingIn = value;
```

Comment:
No point passing in a value here. `setLoggingIn` should just take the state argument, then set `loggingIn` to `true`.

Reply:
Then set `state.loggingIn` to `false` in `setSession` :)

Reply:
@montyanderson, I see. But how about when there is a problem on the login and the section where `setSession` is called is never reached?

Reply:
@gabedealmeida I think this is where error handling comes in. My goal here is to make the mutations as inflexible as possible to reduce the error surface.

For me, it makes sense to have three mutations: `startLogin()`, `successLogin(email...)`, `failLogin(error)`. `isLoggingIn` should be set as a derivative of these mutations, not on it's own.

Reply:
Put another way, we should treat actions as effectively hostile. They have a much larger scope for failure due to concurrency, race conditions, and complexity. Mutations need to provide the most limited possible API for actions to function.